### PR TITLE
Fix: Clearing warnings from console

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
Resolves <https://github.com/tuist/tuist/issues/5552>
/claim #5552 

### Short description 📝

removing unwanted warnings 

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by DhairyaMajmudar** [comment]
Hi @fortmarek can you pls. help in passing workflow and PR review, Thank you : )

**Comment by ollieatkinson** [comment]
@DhairyaMajmudar I would recommend writing some unit tests to prove the behaviour is working and correct

**Comment by DhairyaMajmudar** [comment]
> @DhairyaMajmudar I would recommend writing some unit tests to prove the behaviour is working and correct

Correct! I have no knowledge of writing unit tests can you pls. help in this

**Comment by DhairyaMajmudar** [comment]
@fortmarek  kind ping for PR review

File: Sources/TuistLoader/Models+ManifestMappers/ResourceFileElement+ManifestMapper.swift [hunk 1]
```diff
@@ -24,40 +24,23 @@ extension XcodeGraph.ResourceFileElement {
...
                 .filter { !excluded.contains($0) }
 
-            if files.isEmpty {
-                if FileHandler.shared.isFolder(path) {
-                    logger.warning("'\(path.pathString)' is a directory, try using: '\(path.pathString)/**' to list its files")
-                } else {
-                    // FIXME: This should be done in a linter.
-                    logger.warning("No files found at: \(path.pathString)")
-                }
+            if files.isEmpty && FileHandler.shared.isFolder(path) {
```
Comment (pepicrft) [inline Sources/TuistLoader/Models+ManifestMappers/ResourceFileElement+ManifestMapper.swift:37]: Won't `FileHandler.shared.isFolder(path)` always yield `false` here assuming that `path` is a glob at this point?

File: Sources/TuistLoader/Models+ManifestMappers/CopyFileElement+ManifestMapper.swift [hunk 2]
```diff
@@ -18,38 +18,20 @@ extension XcodeGraph.CopyFileElement {
...
                 .filter(includeFiles)
 
-            if files.isEmpty {
-                if FileHandler.shared.isFolder(path) {
-                    logger.warning("'\(path.pathString)' is a directory, try using: '\(path.pathString)/**' to list its files")
-                } else {
-                    // FIXME: This should be done in a linter.
-                    logger.warning("No files found at: \(path.pathString)")
-                }
+            if files.isEmpty && FileHandler.shared.isFolder(path) {
```
Comment (pepicrft) [inline Sources/TuistLoader/Models+ManifestMappers/CopyFileElement+ManifestMapper.swift:28]: Similar comment here regarding the `path` being a glob at this point.

File: Sources/TuistGenerator/Linter/TargetLinter.swift [hunk 3]
```diff
@@ -337,6 +339,35 @@ class TargetLinter: TargetLinting {
...
+
+        for path in paths {
+            if !FileHandler.shared.exists(path) {
+                lintingIssues.append(LintingIssue(reason: "No files found at: \(path)", severity: .warning))
+                continue
+            }
+
+            if FileHandler.shared.isFolder(path) && !FileHandler.shared.exists(path) {
+                lintingIssues.append(LintingIssue(reason: "No folder found at: \(path)", severity: .warning))
+            }
```
Comment (pepicrft) [inline Sources/TuistGenerator/Linter/TargetLinter.swift:367]:
```suggestion
            if !FileHandler.shared.exists(path) {
                lintingIssues.append(LintingIssue(reason: "No items found at path: \(path.pathString)", severity: .warning))
                continue
            }
```

**Review by pepicrft — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution @DhairyaMajmudar. You can use [these tests](https://github.com/tuist/tuist/blob/main/Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift) as a reference. You'll have to reproduce the scenario in a temporary directory, run the linter, and assert that the lining issues match the ones that you expect.

**Comment by pepicrft** [comment]
@DhairyaMajmudar regarding the failing CI, based on the logs, there seems to be an issue in your changes that cause the compilation to fail. If you clean derived data, run `tuist generate`, and try to build the main workspace, the compilation should fail locally too. Could you try and let me know if that's the case?

**Comment by vijaytholpadi** [comment]
@DhairyaMajmudar I'm looking forward to the fix in this PR. Thanks for taking it up 💯 Can I check if you would be able to followup on the feedback to merge this fix?

**Comment by DhairyaMajmudar** [comment]
> @DhairyaMajmudar I'm looking forward to the fix in this PR. Thanks for taking it up 💯 Can I check if you would be able to followup on the feedback to merge this fix?

Thank you so much @vijaytholpadi for appreciation looking forward to follow up on suggestions today only.

File: Sources/TuistLoader/Models+ManifestMappers/ResourceFileElement+ManifestMapper.swift (hunk 1 above)
Comment (DhairyaMajmudar) [reply Sources/TuistLoader/Models+ManifestMappers/ResourceFileElement+ManifestMapper.swift:37]: Hi @pepicrft Thanks for you review, I'm a bit unsure here can you pls. suggest a fix for this 

cc: @vijaytholpadi

**Comment by vijaytholpadi** [comment]
@DhairyaMajmudar I see [some](https://github.com/tuist/tuist/pull/6420#discussion_r1647114549) open feedback on the PR and Pedro [also mentioned](https://github.com/tuist/tuist/pull/6420#issuecomment-2189433159) that the code is not currently compiling. Were you able to have a look into it?

File: Sources/TuistLoader/Models+ManifestMappers/ResourceFileElement+ManifestMapper.swift (hunk 1 above)
Comment (DhairyaMajmudar) [reply Sources/TuistLoader/Models+ManifestMappers/ResourceFileElement+ManifestMapper.swift]: correct! I've fixed the conditions 👍🏻

File: Sources/TuistLoader/Models+ManifestMappers/CopyFileElement+ManifestMapper.swift (hunk 2 above)
Comment (DhairyaMajmudar) [reply Sources/TuistLoader/Models+ManifestMappers/CopyFileElement+ManifestMapper.swift]: Fixed this as well

**Comment by DhairyaMajmudar** [comment]
@vijaytholpadi  I have done the suggested changes by @pepicrft but a bit unsure for the comment [here](https://github.com/tuist/tuist/pull/6420#issuecomment-2189433159) can you help by pushing the corrections.

**Comment by DhairyaMajmudar** [comment]
@vijaytholpadi, @pepicrft a friendly ping for PR review.

**Comment by pepicrft** [comment]
@DhairyaMajmudar would you mind looking at CI? It seems that the changes in this PR are not compilable.

**Comment by DhairyaMajmudar** [comment]
> @DhairyaMajmudar would you mind looking at CI? It seems that the changes in this PR are not compilable.

A bit confused for the CI, help is much appreciated

**Comment by pepicrft** [comment]
@DhairyaMajmudar CI is failing because the changes in this PR don't compile.

**Review by pepicrft — COMMENTED** [review]
This PR has been inactive for some time, so I'm closing it.

**Pull request closed without merging**